src/test/kotlin/org/rust/ide/hints/RsExpressionTypeProviderTest.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.hints
+
+import org.intellij.lang.annotations.Language
+import org.rust.ProjectDescriptor
+import org.rust.RsTestBase
+import org.rust.WithStdlibRustProjectDescriptor
+import kotlin.test.assertEquals
+
+
+class RsExpressionTypeProviderTest : RsTestBase() {
+    fun `test simple type`() {
+        doTest("""
+        fn main() {
+            let /*caret*/x = 5u32;
+        }
+        """, "u32")
+    }
+
+    @ProjectDescriptor(WithStdlibRustProjectDescriptor::class)
+    fun `test aliased type`() {
+        doTest("""
+        use std::cell::RefCell;
+
+        type BoxedCell<T> = Box<RefCell<T>>;
+
+        fn foo<T>(c: &BoxedCell<T>) {
+            let /*caret*/b = c;
+        }
+        """, "&BoxedCell<T>")
+    }
+
+    private fun doTest(@Language("Rust") code: String, type: String) {
+        InlineFile(code).withCaret()
+
+        val element = myFixture.elementAtCaret
+
+        val provider = RsExpressionTypeProvider()
+        val expressions = provider.getExpressionsAt(element)
+        assertEquals(type, expressions.joinToString(", ") { provider.getInformationHintUnescaped(it) })
```

Comment:
Let's better use `getInformationHint` to ensure that we don't forget to unescape something

Reply:
Ok. I didn't want to use it because then if you have a test failure you see something like "Expected: & lt;Something;& gt". But it's not a big deal.